cli/diag.c

```diff
@@ -126,16 +126,20 @@ static int ltssm_log(int argc, char **argv) {
 		DEVICE_OPTION, PORT_OPTION, {}
 	};
 
-	struct switchtec_diag_ltssm_log output[128];
 	int ret;
 	int port;
-	int log_count = 128;
 	int i;
 
 	ret = diag_parse_common_cfg(argc, argv, CMD_DESC_LTSSM_LOG,
 				    &cfg, opts);
 	if (ret)
 		return ret;
+	
+	int log_count = 512;
+	if (!switchtec_is_gen5(cfg.dev)) {
+		log_count = 128;
+	}
```

Comment:
The surrounding {} should be omitted for this single-statement block inside the if condition, as well as other similar instances.